grammar.js

```diff
@@ -608,6 +609,36 @@ module.exports = grammar({
 
     pointer_type: $ => prec(PREC.POSTFIX, seq($._type, '*')),
 
+    function_pointer_type: $ => seq(
+      'delegate',
+      '*',
+      optional($.calling_convention_specifier),
+      '<',
+      commaSep1($.funcptr_parameter),
+      '>'
+    ),
+
+    calling_convention_specifier: $ => choice(
+      'managed',
+      seq(
+        'unmanaged',
+        optional(seq('[', $.unmanaged_calling_convention, ']'))
+      )
+    ),
+
+    unmanaged_calling_convention: $ => choice(
```

Comment:
Roslyn definitely seems to vary from the proposal grammar on this one. They have line 625 as a rule called `FunctionPointerUnamangedCallingConventionList` and then each convention inside is a `FunctionPointerUnmanagedCallingConvention`.